src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -502,6 +502,21 @@ protected DynamoModel(IStartConfiguration config)
                 }
             }
 
+            // At this point, pathManager.PackageDirectories only has 1 element which is the directory
+            // in AppData. If list of PackageFolders is empty, add the folder in AppData to the list since there
+            // is no additional location specified. Otherwise, update pathManager.PackageDirectories to include
+            // PackageFolders
+            if (PreferenceSettings.PackageFolders.Count == 0)
+                PreferenceSettings.PackageFolders = new List<string>(pathManager.PackagesDirectories);
+            else
+                pathManager.LoadPackageFolders(PreferenceSettings.PackageFolders);
```

Comment:
During start-up, `PreferenceSettings.PackageFolders` may or may not contain any entry (in which case, the user has not specified other preferred location to load packages from, before this session). When it is empty, the default `AppData` location will always be used, which means the `pathManager.PackagesDirectories` is always guaranteed to have at least `AppData` entry. In this scenario, the above `if` statement populates the `PreferenceSettings.PackageFolders` with the one-and-only default `AppData` entry. Otherwise (if the `PreferenceSettings.PackageFolders` comes preloaded with some entries in it), the `pathManager.PackagesDirectories` will be overwritten with the entries coming from `PreferenceSettings.PackageFolders`. Does that make sense?